src/automatic.ts

```diff
@@ -28,60 +28,158 @@ import {Server} from './server';
 import {specUrl} from './specs';
 import {wait} from './util';
 
-function combineResults(results: BenchmarkResult[]): BenchmarkResult {
-  const combined: BenchmarkResult = {
-    ...results[0],
-    millis: [],
-  };
-  for (const result of results) {
-    combined.millis.push(...result.millis);
-  }
-  return combined;
-}
-
 interface Browser {
   name: string;
   driver: webdriver.WebDriver;
   initialTabHandle: string;
 }
 
-export async function automaticMode(
-    config: Config, servers: Map<BenchmarkSpec, Server>):
-    Promise<Array<ResultStatsWithDifferences>|undefined> {
-  let completeGithubCheck;
-  if (config.githubCheck !== undefined) {
-    completeGithubCheck = await github.createCheck(config.githubCheck);
+export class AutomaticMode {
+  private config: Config;
```

Comment:
Many of these fields can be made readonly as well

Reply:
Done.